int32 id
int32 aid
int32 type
int32 linkid
